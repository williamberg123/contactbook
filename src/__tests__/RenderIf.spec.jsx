import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import RenderIf from '../components/RenderIf';

describe('<RenderIf />', () => {
	it('should render element when prop isTrue is true', () => {
		render(
		<RenderIf isTrue={5 > 3}>
			<p>Fui renderizado</p>
		</RenderIf>,
		);

		expect(screen.getByText(/fui renderizado/i)).toBeInTheDocument();
	});

	it('should not render element when prop isTrue is false', () => {
		render(
		<RenderIf isTrue={5 < 3}>
			<p>Fui renderizado</p>
		</RenderIf>,
		);

		expect(screen.queryByText(/fui renderizado/i)).not.toBeInTheDocument();
	});

	it('should match snapshot', () => {
		render(
		<RenderIf isTrue>
			<p>Fui renderizado</p>
		</RenderIf>,
		);

		expect(screen.queryByText(/fui renderizado/i)).toMatchSnapshot();
	});
});
