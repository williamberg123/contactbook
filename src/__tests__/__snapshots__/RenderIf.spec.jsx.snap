// Vitest Snapshot v1

exports[`<RenderIf /> > should match snapshot 1`] = `
<p>
  Fui renderizado
</p>
`;
